### PR TITLE
Rain Water Trapping Problem in C++

### PR DESCRIPTION
**Pull request by adityaiiitL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes, I have read all the contributing guidelines on Pull requests

### Description

In this I have solved a problem in which we have to find out the total units of water trapped in-between the buildings, we are taking input of number of test-cases and in each test-case we are taking n as an input that denotes the number of buildings and the next line is regarding taking input of heights of n buildings. I have solved this problem using an optimized method in C++ with complexity of O(n).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#3955

### CODE REVIEW

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+*/
+#include<bits/stdc++.h>
+using namespace std;
+
+// Function to return amount of water trapped
+void solve()
+{
+    long long int no_of_buildings;
+    //taking input of number of buildings
+    cin>>no_of_buildings;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp:22]: Use cout to display message how you are reading inputs

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    // Making an array containing heights of buildings
+    long long int height[no_of_buildings];
+    for(long long int i=0;i<no_of_buildings;i++){
+        
+        //Taking input of height of ith-building
+        cin>>height[i];
+    }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp:30]: Use cout to display message how you are reading inputs

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/rain_water_trapping.cpp (hunk 2 above)
Comment (adityaiiitL) [reply C-Plus-Plus/cp/rain_water_trapping.cpp:30]: I have done the required changes, please review.

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    //taking input of number of buildings
+    cin>>no_of_buildings;
+    // Making an array containing heights of buildings
+    long long int height[no_of_buildings];
+    for(long long int i=0;i<no_of_buildings;i++){
+        
+        //Taking input of height of ith-building
+        cout<<"Height of building: ";
+        cin>>height[i];
+    }
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp]:
```
cout<<"Height of building: ";
for(long long int i=0;i<no_of_buildings;i++){

        //Taking input of height of ith-building       
        cin>>height[i];
}
```

Rewrite it,

File: C-Plus-Plus/cp/rain_water_trapping.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    Input:
+    5
+    3 0 2 0 4
+    ---------------
+    Output:
+    7
+    ---------------
+    Time-Complexity: O(n)
+    Space-Compleixty: O(1) 
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/rain_water_trapping.cpp]: Left align it and write the output in the way your code is producing

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/rain_water_trapping.cpp (hunk 4 above)
Comment (adityaiiitL) [reply C-Plus-Plus/cp/rain_water_trapping.cpp]: I have done the required changes, please check and review.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**